File: 0055-jump-game/0055-jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int curr = nums.size() - 1;
        for (int i = nums.size() - 2; i >= 0; i--) {
            if (i + nums[i] >= curr) { curr = i; }
        }
        return (curr == 0);
    }
};

File: 0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        curr = len(nums) - 1
        for i in range(len(nums) - 1, -1, -1):
            if i + nums[i] >= curr:
                curr = i
                
        return True if curr == 0 else False